tensorflow_addons/layers/normalizations.py

```diff
@@ -316,3 +316,225 @@ def __init__(self, **kwargs):
+        self.gamma_regularizer = tf.keras.regularizers.get(gamma_regularizer)
+        self.tau_regularizer = tf.keras.regularizers.get(tau_regularizer)
+        self.beta_constraint = tf.keras.constraints.get(beta_constraint)
+        self.gamma_constraint = tf.keras.constraints.get(gamma_constraint)
+        self.tau_constraint = tf.keras.constraints.get(tau_constraint)
+        self.eps_var_initializer = tf.keras.initializers.get(
+            eps_var_initializer)
+        self.supports_masking = True
+        if isinstance(axis, list):
+            self.axis = axis[:]
+        else:
+            raise TypeError('axis must be list, type given: %s' % type(axis))
+        self._check_axis()
+
+    def build(self, input_shape):
+        self._check_if_input_shape_is_none(input_shape)
+        self._create_input_spec(input_shape)
+        self._add_gamma_weight(input_shape)
+        self._add_beta_weight(input_shape)
+        self._add_tau_weight(input_shape)
+        self._add_eps_var_weight()
+        self.built = True
+        super(FilterResponseNormalization, self).build(input_shape)
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
+
+    def get_config(self):
+        config = {
+            'axis':
+            self.axis,
+            'epsilon':
+            self.epsilon,
+            'beta_initializer':
+            tf.keras.initializers.serialize(self.beta_initializer),
+            'gamma_initializer':
+            tf.keras.initializers.serialize(self.gamma_initializer),
+            'tau_initializer':
+            tf.keras.initializers.serialize(self.tau_initializer),
+            'eps_var_initializer':
+            tf.keras.initializers.serialize(self.eps_var_initializer),
+            'beta_regularizer':
+            tf.keras.regularizers.serialize(self.beta_regularizer),
+            'gamma_regularizer':
+            tf.keras.regularizers.serialize(self.gamma_regularizer),
+            'tau_regularizer':
+            tf.keras.regularizers.serialize(self.tau_regularizer),
+            'beta_constraint':
+            tf.keras.constraints.serialize(self.beta_constraint),
+            'gamma_constraint':
+            tf.keras.constraints.serialize(self.gamma_constraint),
+            'tau_constraint':
+            tf.keras.constraints.serialize(self.tau_constraint)
+        }
+        base_config = super(FilterResponseNormalization, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+    def call(self, inputs):
+        # Compute the mean norm of activations per channel.
+        nu2 = tf.reduce_mean(tf.square(inputs), axis=self.axis, keepdims=True)
+        # Perform FRN.
+        inputs = inputs * tf.math.rsqrt(nu2 + tf.math.abs(self.epsilon))
+        # Return after applying the Offset-ReLU non-linearity.
+        return tf.maximum(self.gamma * inputs + self.beta,
+                          self.tau + self.eps_var)
```

Comment:
As suggested, there shouldn't be an eps in TLU.

In line with my earlier comment, I would suggest making TLU its own layer. Further, I would suggest that the implementation also include a configuration option to produce Affine-TLU (ref. paper). While Affine-TLU was shown to be similar/worse than TLU for image classification, it may still be useful in as yet unexplored applications.

Reply:
Yeah, It's a typo actually. I will remove it

Reply:
@saurabhme `gamma` and `beta` would be part of `FRN` and not TLU. For TLU, only `tau` is required. Rest everything is part of FRN. Just trying to clear it before I modify this PR. Thanks

Reply:
Yes. That is correct.

Reply:
Thanks for the clarification

Reply:
@saurabhme Sorry to  bug you again but if we implement TLU as a separate layer, how is it diffferent from PReLU? To me, it seems the same until unless I am missing something

Reply:
PReLU computes
`max(x, ax)`

While TLU computes 
`max(x, t)`

While Affine-TLU computes
`max(x, ax+t)`

Where a, t are learnable parameters. The difference is in the form of function that threshold takes. In PReLU it is a scaling of x, in TLU it is a bias while in Affine-TLU it is a affine function of x.

Reply:
Cool. Thanks